extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -67,14 +67,22 @@
     private YoutubeParsingHelper() {
     }
 
-    private static final String HARDCODED_CLIENT_VERSION = "2.20200214.04.00";
-    private static String clientVersion;
+    public static final String YOUTUBEI_V1_URL = "https://www.youtube.com/youtubei/v1/";
 
+    private static final String HARDCODED_CLIENT_VERSION = "2.20210601.07.00";
+    private static final String HARDCODED_KEY = "AIzaSyAO_FJ2SlqU8Q4STEHLGCilw_Y9_11qcW8";
+    private static final String[] MOBILE_YOUTUBE_KEYS = {"AIzaSyA8eiZmM1FaDVjRy-df2KTyQ_vz_yYM39w",
+            "16.20.35"};
+    private static String clientVersion;
     private static String key;
 
-    private static final String[] HARDCODED_YOUTUBE_MUSIC_KEYS = {"AIzaSyC9XL3ZjWddXya6X74dJoCTL-WEYFDNX30", "67", "0.1"};
+    private static final String[] HARDCODED_YOUTUBE_MUSIC_KEYS =
```

Comment:
I think this name is misleading. At first I interpreted it as `an array of possible keys`, while instead it is `multiple parameters for the same key`. I would just remove the `s`. Also above for `MOBILE_YOUTUBE_KEYS`.
```suggestion
    private static final String[] HARDCODED_YOUTUBE_MUSIC_KEYS =
```

Reply:
I didn't change this, it was as it is now before I made changes to YoutubeParsingHelper, but I applied your suggestion and split the `MOBILE_YOUTUBE_KEYS` into two strings: `MOBILE_YOUTUBE_KEY` and `MOBILE_YOUTUBE_CLIENT_VERSION` because there is no mean to extract they so we will always use these keys as they are.